unblob/handlers/compression/gzip.py

```diff
@@ -38,12 +40,16 @@ class GZIPHandler(Handler):
         // compression method (0x8 = DEFLATE)
         // flags, 00011111 (0x1f) is the highest since the first 3 bits are reserved
         // unix time
-        // eXtra FLags
-        // Operating System (RFC1952 describes 0-13, or 255)
-        $gzip_magic = /\x1f\x8b\x08[\x00-\x1f][\x00-\xff]{4}[\x00-\x04][\x00-\x0c\xff]/
+        // eXtra FLags (2 or 4 per RFC1952 2.3.1)
+        // Operating System (0-13, or 255 per RFC1952 2.3.1)
+        $gzip_magic = {{ 1F 8B 08 (00 | 01 | 02 | 03 | 04 | 05 | 06 | 07 | 08 | 09 | 0A | 0B | 0C | 0D | 0E | 0F | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 1A | 1B | 1C | 1D | 1E) [4] (02 | 04) (00 | 01 | 02 | 03 | 04 | 05 | 06 | 07 | 08 | 09 | 0A | 0B | 0C | 0D | FF) }}
     condition:
-        $gzip_magic
-    """
+        $gzip_magic and
+        // timestamp is before 2039
+        uint32(@gzip_magic+4) < {timestamp}
```

Comment:
Uhm, the comment doesn't agree with the code here. Is there a real benefit by restricting the timestamp to not be in the future? What happens with time zones?

Reply:
I've thought about it and just removed the timestamp check, the gzip_magic should be more than sufficient. I've seen gzip files with timestamp in the future so better not check that field.